clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/AggKafkaConsumerService.java

```diff
@@ -115,20 +122,21 @@ private KafkaConsumerService getKafkaConsumerService(final String kafkaURL) {
    * @param consumerProperties consumer properties that are used to create {@link KafkaConsumerService}
    */
   public synchronized KafkaConsumerService createKafkaConsumerService(final Properties consumerProperties) {
-    final String kafkaUrl = consumerProperties.getProperty(KAFKA_BOOTSTRAP_SERVERS);
+    String kafkaUrl = consumerProperties.getProperty(KAFKA_BOOTSTRAP_SERVERS);
     Properties properties = sslPropertiesSupplier.get(kafkaUrl).toProperties();
     consumerProperties.putAll(properties);
     if (kafkaUrl == null || kafkaUrl.isEmpty()) {
       throw new IllegalArgumentException("Kafka URL must be set in the consumer properties config. Got: " + kafkaUrl);
     }
-    final KafkaConsumerService alreadyCreatedConsumerService = kafkaServerToConsumerServiceMap.get(kafkaUrl);
+    String resolvedKafkaUrl = kafkaClusterUrlResolver == null ? kafkaUrl : kafkaClusterUrlResolver.apply(kafkaUrl);
+    final KafkaConsumerService alreadyCreatedConsumerService = getKafkaConsumerService(resolvedKafkaUrl);
```

Comment:
Looks good overall. Just one minor concern:
Does the resolver return our preferred url? Which is the IPv6 url?
I hope our codes can be defensive to this case - let's say IPv6 url is the one that can be resolved in future, and one day the IPv4 url doesn't work anymore; I hope the system will keep the consumers that use the new url.

Reply:
Yes resolver will resolve to a primary url (`url` field in `kafkaClusterMap`).

Reply:
Sounds good! Let's make sure IPv6 is the primary url.

Reply:
Ack.